src/core/trtserver2.h

```diff
@@ -297,6 +298,25 @@ TRTSERVER_EXPORT TRTSERVER_Error* TRTSERVER2_InferenceRequestOutputData(
     const void** base, size_t* byte_size, TRTSERVER_Memory_Type* memory_type,
     int64_t* memory_type_id);
 
+/// Get the indices of all unique models in the model repository.
+/// The caller does not own the returned strings and must not modify or
+/// delete them. The lifetime of the returned strings extends only as
+/// long as 'server' and must not be accessed once 'server' is deleted.
+/// \param server The inference server object.
+/// \param model_indices Returns the TRTSERVER2_ModelIndex object that
+/// can then be used to get the indices of all unique models in the model
+/// repository.
+/// \return a TRTSERVER_Error indicating success or failure.
+TRTSERVER_EXPORT TRTSERVER_Error* TRTSERVER2_ServerModelRepositoryIndexNew(
```

Comment:
The only function you need here is:
TRTSERVER_EXPORT TRTSERVER_Error* TRTSERVER2_ServerModelIndex(
    TRTSERVER_Server* server, TRTSERVER_ModelIndex** index);